pkg/upgrade/upgrades/schema_changes.go

```diff
@@ -411,3 +412,28 @@ func onlyHasColumnFamily(
 	}
 	return true, nil
 }
+
+// bumpSystemDatabaseSchemaVersion bumps the SystemDatabaseSchemaVersion
+// field for the system database descriptor. It should be called at the end
+// of any upgrade that creates or modifies the schema of any system table.
+func bumpSystemDatabaseSchemaVersion(
+	ctx context.Context, cs clusterversion.ClusterVersion, d upgrade.TenantDeps,
+) error {
+	return d.DB.DescsTxn(ctx, func(ctx context.Context, txn descs.Txn) error {
```

Comment:
should this include a WaitForOneVersion, so that we know anyone who observes the cluster version changed -- which will be done after the upgrade func this is called in returns -- won't also still have a lease of the old db desc?

Reply:
This is already implicitly done in `DescsTxn`!

https://github.com/cockroachdb/cockroach/blob/173a6db0eacdcc0de72fd2b8542a9c6662deda5e/pkg/sql/internal.go#L1668
https://github.com/cockroachdb/cockroach/blob/173a6db0eacdcc0de72fd2b8542a9c6662deda5e/pkg/sql/internal.go#L1708

Reply:
Oh, fancy! I've been out of the loop for awhile here I guess.